src/app/api/v1/bank-account/delete/route.ts

```diff
@@ -0,0 +1,30 @@
+import { checkUserAuthOrThrowError } from "@/app/api/v1/server-actions";
+import bankAccountService from "@/domain/bankAccount-domain/ba-service";
+import { NextResponse } from "next/server";
+
+export async function DELETE(request: Request) {
+  try {
+    const user = await checkUserAuthOrThrowError(request);
+    if ("error" in user) {
+      return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+    }
+
+    const { searchParams } = new URL(request.url);
+    const bankAccountId = searchParams.get("id");
+
+    if (!bankAccountId) {
+      return NextResponse.json({ error: "Missing bank account ID" }, { status: 400 });
+    }
+    const result = await bankAccountService.deleteBankAccount(bankAccountId);
+
+    await bankAccountService.deleteBankAccount(bankAccountId);
```

Comment:
This is unnecessary - this will throw error everytime, because you are doing this operation on line 18 and here on line 20 :) 
```
line 18 - delete it once
line 20 - trying to delete it again, but it is not in DB -> error
```